[MDB IGNORE] Shuttle engines part 2: Engines are now machines

JohnFulpWillard opened the pull request:
## About The Pull Request

Engines are no longer structures, instead they're machines

Now they (except large/huge ones) have circuit boards, but they can't be printed by default. Instead, this allows the machines to be destroyed and rebuilt, but maybe in the future we can add these as printable machines.
Currently the only way to get more is through Cargo so maybe we can add the boards there instead? This is really just opening the door to anyone who wishes to expand upon this.

I also removed 2 engine types, 'router' and 'platform', beacuse their sprites sucked, they had no unique mechanics, and cutting down on the amount of engines might help make the others unique just by having less to work with. They also weren't engines at all, despite what they're subtypes of.

## Why It's Good For The Game

These are already technically machines since they affect Shuttles, I thought it might be a good idea for them to be turned into machines so maybe in the future, unique functionalities can be added to them.

## Changelog

:cl:
add: Engines are now machines, not structures. If destroyed, they will drop circuit boards so you can rebuild them.
removal: In-wall engines have been remvoed. The Tram now uses full-blown engines.
/:cl:

JohnFulpWillard committed:
Makes engines machines instead of structures

JohnFulpWillard committed:
Updates the maps

JohnFulpWillard committed:
Fixes boards and anchoring

JohnFulpWillard committed:
Removes 2 unused engine types

Router was actually used a total of once, so I just replaced it with propulsion.
I think cutting down on these useless engine types that make no difference in-game would be a nice first step to adding more functionalities to them.

JohnFulpWillard committed:
Don't use power (since shuttles dont have)

Shuttles don't have APCs, instead they just have infinite power, so I'm removing their power usage for now. I'm hoping this can be removed when unique mechanics are added to engines, because I would like them to make use of power like other machines.

JohnFulpWillard committed:
re-organizes vars

JohnFulpWillard committed:
deletes deleted dm file

private-tristan commented:
will they be able to be screwdriver+crowbar'd?

Dmeto commented:
would like to see them added to cargo fab, incase of the classic syndy bomb event. just make sure they aren't deleted as structures are ignored by supply shuttle's delete and the last time I checked machines weren't.

JohnFulpWillard committed:
Slightly improves cargo selling code

JohnFulpWillard commented:
> will they be able to be screwdriver+crowbar'd?

No but it is something that can be added easily now.

> Would like to see them added to cargofab, incase of the classic syndy bomb event. 

It should still drop the boards for rebuilding, otherwise I would prefer if they were something you ordered from cargo instead of printing out at a lathe.

> Just make sure they aren't deleted, as structures are ignored by supply shuttle's delete and the last time I checked machines weren't.

The cargo shuttle sells everything on it that isn't a camera or unanchored, Shuttles start anchored so it's fine.

CursedBirb commented:
Ordering them through cargo might even make sense as i dont think Nanotrasen would like to see their employes potentially dangerous crafts

ShizCalev commented:
Conflicto

JohnFulpWillard committed:
Merge branch 'master' into engines-part-two

san7890 commented on tools/UpdatePaths/Scripts/shuttle_engines.txt:
```diff
@@ -0,0 +1,8 @@
+/obj/structure/shuttle/engine/@SUBTYPES : /obj/machinery/power/shuttle_engine/@SUBTYPES{@OLD}
```
update the name of this file to have the number of this PR so it sorts correctly?

JohnFulpWillard committed:
Renames the updatepaths

JohnFulpWillard commented:
me when i mark map edits as viewed
![image](https://user-images.githubusercontent.com/53777086/191692834-a24de3b3-eb54-450f-a3db-dc94d0903c12.png)

san7890 commented on code/game/shuttle_engines.dm:
```diff
@@ -109,95 +108,88 @@
-	name = "engine router"
-	icon_state = "router"
-	desc = "Redirects around energized particles in engine structures."
+/obj/machinery/power/shuttle_engine/propulsion/in_wall
```
In_wall propulsion engines feel distinct enough to me that they shouldn't use the same circuitboard as standard propulsion engines (or at least, if you were to re-assemble it, you would get the full-sized propulsion engine anyways). Right?

JohnFulpWillard replied:
why do in-wall engines even exist bro i hate this place

JohnFulpWillard committed:
Removes in_wall engines

I hate this stupid engine it sucks it's useless it's used solely for the tram it provides nothing of benefit to the server
replaces them with regular engines

JohnFulpWillard commented:
removed in wall engines
tram looks like this now
![image](https://user-images.githubusercontent.com/53777086/192105835-19331319-7708-451d-91a2-1b33645d5473.png)

i decided to make hilbert hotel look a little silly too
![image](https://user-images.githubusercontent.com/53777086/192105849-05405af3-6a02-436a-ac2f-16c56a1c0f36.png)

JohnFulpWillard committed:
Merge branch 'master' into engines-part-two

Dmeto commented:
@Fikou 
As this now touches /ruins/lavaland

JohnFulpWillard commented:
this is demented

JohnFulpWillard committed:
Merge branch 'master' into engines-part-two
